### PR TITLE
Added support for water-only and water-only + thermal to flow.

### PR DESCRIPTION
**Pull request by blattms**
Needs OPM/opm-tests#753

### CODE REVIEW

**Comment by blattms** [comment]
jenkins build this please

**Comment by blattms** [comment]
@goncalvesmachadoc please check this.

**Comment by akva2** [comment]
Could consider changing tests to run through flow binary (and perform manual renames in opm-tests)

File: flow/flow_ebos_onephase_energy.cpp [hunk 1]
```diff
@@ -60,7 +60,34 @@ struct Indices<TypeTag, TTag::EclFlowProblemSimple>
 
 } // namespace Opm::Properties
 
-int flowEbosOnephaseEnergyMain(int argc, char** argv)
+void flowEbosWaterEnergyOnlySetDeck(double setupTime, std::shared_ptr<Deck> deck,
```
Comment (goncalvesmachadoc) [inline flow/flow_ebos_onephase_energy.cpp:63]: Shouldn't be 
` void flowEbosWaterOnlyEnergySetDeck`
instead of 
` void flowEbosWaterEnergyOnlySetDeck`
?

Comment (blattms) [reply flow/flow_ebos_onephase_energy.cpp:63]: Indeed. Good catch.

File: opm/simulators/flow/Main.hpp [hunk 2]
```diff
@@ -35,6 +35,7 @@
 #include <flow/flow_ebos_brine.hpp>
 #include <flow/flow_ebos_brine_saltprecipitation.hpp>
 #include <flow/flow_ebos_gaswater_saltprec_vapwat.hpp>
+#include <flow/flow_ebos_onephase.hpp>
```
[the remark this answers is not in the document]
Comment (blattms) [reply opm/simulators/flow/Main.hpp:38]: Yep.

**Comment by blattms** [comment]
> .... (and perform manual renames in opm-tests)

Not sure what you mean by that. Once you tell me, I will.

**Comment by akva2** [comment]
```
akva@SINTEFPC9321:~/kode/opm/opm-tests/spe1$ ls opm-simulation-reference/flow_onephase
SPE1CASE1_WATER.EGRID  SPE1CASE1_WATER.INIT  SPE1CASE1_WATER.SMSPEC  SPE1CASE1_WATER.UNRST  SPE1CASE1_WATER.UNSMRY
```

the simulator used is recorded in the directory the reference data is stored in. so you can't just change the test to use the 'flow' binary if you want to retain the old reference data.

**Comment by akva2** [comment]
also need to update tests/update_reference_data.sh

**Comment by blattms** [comment]
@akva2 Done. Compared to you I must be asleep the whole day. Amazing reaction time.

**Comment by blattms** [comment]
jenkins build this opm-tests=753 please

**Comment by blattms** [comment]
:flushed: One of these days I would loose my math degree if there wasn't  `git commit --amend` to hide stupid mistakes.

**Comment by blattms** [comment]
jenkins build this opm-tests=753 please

**Comment by blattms** [comment]
@akva2 Please double check my changes to tests/update_reference_data.sh.I don't trust myself right now

**Comment by akva2** [comment]
those changes look fine.

**Comment by blattms** [comment]
Thanks.
At least the test cases run through now. Need to dive in and see what stupid mistake I did now (maybe moved and old version in the other PR...).

File: opm/simulators/flow/Main.hpp [hunk 3]
```diff
@@ -652,6 +664,36 @@ class Main
...
+    int runWaterOnlyEnergy(const Phases& phases)
+    {
+        if (!phases.active(Phase::WATER) || phases.size() != 2) {
+            if (outputCout_)
+                std::cerr << "No valid configuration is found for water-only simulation, valid options include "
+                          << "water, water + thermal" << std::endl;
+
+            return EXIT_FAILURE;
+        }
+        flowEbosWaterOnlySetDeck(
```
Comment (akva2) [inline opm/simulators/flow/Main.hpp:691]: wrong set deck function, means wrong typetag settings means no TEMP output enabled.

**Comment by blattms** [comment]
jenkins build this opm-tests=753 please

**Review by akva2 — APPROVED** [review]

**Comment by blattms** [comment]
🥵, but finally there.

**Pull request merged**